### PR TITLE
Issue 3599: Part2 - Update documentation about security config parameters

### PR DESCRIPTION
**Pull request by ravisharda**
**Change log description**  
This PR updates security configuration parameters documentation to reflect the latest set of parameters and (hopefully) describes the parameters better.

**Purpose of the change**  
Addresses part 2 of issue #3599.

**What the code does**  
* Adds some missing params
* Removes/update the ones that have changed
* Updates descriptions with more details
* Organizes the document into two parts - distributed and standalone deployment mode
* Adds an additional column to reflect the pertinent feature 

**How to verify it**  
(Optional: steps to verify that the changes are effective)

### CODE REVIEW

**Comment by ravisharda** [comment]
I just realized there is an issue with the branch using which this PR was submitted. Please hold off on the review.

**Comment by ravisharda** [comment]
This PR was closed by mistake. Reopening it.

**Comment by ravisharda** [comment]
I've resolved the issue with the branch. The PR is not ready for review.

File: documentation/src/docs/security/pravega-security-configurations.md [hunk 1]
```diff
@@ -9,46 +9,65 @@ You may obtain a copy of the License at
...
-|Parameter|Details|Default Value|
-|---------|-------|-------------|
-|singlenode.enableTls|Enable TLS between all the components deployed within the singlenode| False|
-|singlenode.enableAuth|Enable authentication/authorization between all the components within the singlenode |False|
-|singlenode.certFile|If TLS is enabled, the public key certificate is used for internal communication between segment store and controller|Empty|
-|singlenode.keyFile|If TLS is enabled, this represents the private key by all the server sockets| Empty|
-|singlenode.passwdFile|If _auth_ is enabled, this represents the password file for the default _auth_ plugin implementation|Empty|
-|singlenode.userName|If _auth_ is enabled, this represents the username used for internal communication between segment store and controller|Empty|
-|singlenode.passwd|If _auth_ is enabled, this represents the password used for internal communication between segment store and controller|Empty|
+This document describes the security configuration parameters for Pravega, in both distributed and standalone modes.
```
Comment (Aparnarr) [inline documentation/src/docs/security/pravega-security-configurations.md]: **distributed** and **standalone**

File: documentation/src/docs/security/pravega-security-configurations.md [hunk 2]
```diff
@@ -9,46 +9,65 @@ You may obtain a copy of the License at
...
+
+## Security Configuration Parameters in Distributed Mode
+
+In the distributed mode, Controllers and Segment Stores are configured individually. The following sub-sections describe
+their Transport Layer Security (TLS) and Auth (Authentication and Authorization) parameters.
+
+
+### Segment store <a name = "pravega-segment-store"></a>
+
```
Comment (Aparnarr) [inline documentation/src/docs/security/pravega-security-configurations.md]: <a name....> not required.......may i know wat's the purpose of it.?...i can help u

File: documentation/src/docs/security/pravega-security-configurations.md [hunk 3]
```diff
@@ -9,46 +9,65 @@ You may obtain a copy of the License at
...
+|Parameter|Description|Default Value|Feature|
+|---------|-------|-------------|------------|
+| pravegaservice.enableTls | Whether to enable TLS for client-server communications. | False | TLS |
+| pravegaservice.certFile | Path of the X.509 PEM-encoded server certificate file for the service. | Empty | TLS |
+| pravegaservice.keyFile | Path of the PEM-encoded private key file for the service. | Empty | TLS |
+| pravegaservice.secureZK | Whether to enable TLS for communication with Apache Zookeeper. | False | TLS |
```
Comment (Aparnarr) [inline documentation/src/docs/security/pravega-security-configurations.md]: all the parameters inside         `............`  (format)

File: documentation/src/docs/security/pravega-security-configurations.md [hunk 4]
```diff
@@ -9,46 +9,65 @@ You may obtain a copy of the License at
...
+| autoScale.tlsEnabled | Whether to enable TLS for internal communication with the Controllers. | False | TLS |
+| autoScale.tlsCertFile | Path of the PEM-encoded X.509 certificate file used for TLS connections with the Controllers. | Empty | TLS |
+| autoScale.validateHostName | Whether to enable hostname verification for TLS connections with the Controllers. | True | TLS |
+| autoScale.authEnabled | Whether to enable authentication and authorization for internal communications with the Controllers. | False | Auth |
+| autoScale.tokenSigningKey | The key used for signing the delegation tokens. | Empty | Auth |
+| bookkeeper.tlsEnabled | Whether to enable TLS for communication with Apache Bookkeeper. | False | TLS |
+| bookkeeper.tlsTrustStorePath | Path of the truststore file in `.jks` format for TLS connections with Apache Bookkeeper. | Empty | TLS |
+
+
+### Controller <a name ="pravega-controller"></a>
```
Comment (Aparnarr) [inline documentation/src/docs/security/pravega-security-configurations.md]: remove <a...

File: documentation/src/docs/security/pravega-security-configurations.md [hunk 5]
```diff
@@ -9,46 +9,65 @@ You may obtain a copy of the License at
...
+
+|Parameter|Details|Default Value|Feature|
```
Comment (Aparnarr) [inline documentation/src/docs/security/pravega-security-configurations.md]: parameters inside `      `

File: documentation/src/docs/security/pravega-security-configurations.md [hunk 6]
```diff
@@ -9,46 +9,65 @@ You may obtain a copy of the License at
...
+|---------|-------|-------------|-------|
+| controller.auth.tlsEnabled | Whether to enable TLS for client-server communication. | False | TLS |
+| controller.auth.tlsCertFile | Path of the X.509 PEM-encoded server certificate file for the service. | Empty | TLS |
+| controller.auth.tlsKeyFile | Path of the PEM-encoded private key file for the service. | Empty | TLS |
+| controller.auth.tlsTrustStore | Path of the PEM-encoded truststore file for TLS connections with Segment Stores. | Empty | TLS |
```
Comment (Aparnarr) [inline documentation/src/docs/security/pravega-security-configurations.md]: PEM format

File: documentation/src/docs/security/pravega-security-configurations.md [hunk 7]
```diff
@@ -9,46 +9,65 @@ You may obtain a copy of the License at
...
+their Transport Layer Security (TLS) and Auth (Authentication and Authorization) parameters.
+
+
+### Segment store <a name = "pravega-segment-store"></a>
+
+|Parameter|Description|Default Value|Feature|
+|---------|-------|-------------|------------|
+| pravegaservice.enableTls | Whether to enable TLS for client-server communications. | False | TLS |
+| pravegaservice.certFile | Path of the X.509 PEM-encoded server certificate file for the service. | Empty | TLS |
+| pravegaservice.keyFile | Path of the PEM-encoded private key file for the service. | Empty | TLS |
```
Comment (Aparnarr) [inline documentation/src/docs/security/pravega-security-configurations.md]: PEM format...

File: documentation/src/docs/security/pravega-security-configurations.md [hunk 8]
```diff
@@ -9,46 +9,65 @@ You may obtain a copy of the License at
...
+| pravegaservice.secureZK | Whether to enable TLS for communication with Apache Zookeeper. | False | TLS |
+| pravegaservice.zkTrustStore | Path of the truststore file in `.jks` format for TLS connections with Apache Zookeeer. | Empty | TLS |
+| pravegaservice.zkTrustStorePasswordPath | Path of the file containing the password of the truststore used for TLS connections with Apache Zookeeper. | Empty | TLS |
+| autoScale.tlsEnabled | Whether to enable TLS for internal communication with the Controllers. | False | TLS |
+| autoScale.tlsCertFile | Path of the PEM-encoded X.509 certificate file used for TLS connections with the Controllers. | Empty | TLS |
```
Comment (Aparnarr) [inline documentation/src/docs/security/pravega-security-configurations.md]: Please check for consistency for `PEM` and `Auth` formats...in all the places

File: documentation/src/docs/security/pravega-security-configurations.md [hunk 9]
```diff
@@ -9,46 +9,65 @@ You may obtain a copy of the License at
...
+| controller.zk.tlsTrustStorePasswordFile | Path of the file containing the password of the truststore used for TLS connections with Apache Zookeeper. | Empty | TLS |
+| controller.auth.enabled | Whether to enable authentication and authorization for clients. | False | Auth |
+| controller.auth.userPasswordFile | Path of the file containing user credentials and ACLs, for the PasswordAuthHandler.| Empty | Auth |
+| controller.auth.tokenSigningKey | Key used to sign the delegation tokens for Segment Stores. | Empty | Auth |
+
+
+## Security Configurations in Standalone Mode
+
+For ease of use, Pravega standalone hides some of the configurations that are mentioned above for configuring
+Segment Store and Controller services individually for deployments in distributed mode.
```
Comment (Aparnarr) [inline documentation/src/docs/security/pravega-security-configurations.md]: Please rephrase

File: documentation/src/docs/security/pravega-security-configurations.md [hunk 10]
```diff
@@ -9,46 +9,65 @@ You may obtain a copy of the License at
...
+|singlenode.enableTls | Whether to enable TLS for client-server communications. | False | TLS |
+|singlenode.certFile | Path of the X.509 PEM-encoded server certificate file for the server. |Empty| TLS |
+|singlenode.keyFile | Path of the PEM-encoded private key file for the service. | Empty | TLS |
+|singlenode.keyStoreJKS | Path of the keystore file in `.jks` for the REST interface. | Empty | TLS |
+|singlenode.keyStoreJKSPasswordFile |Path of the file containing the keystore password for the REST interface. | Empty | TLS |
+|singlenode.trustStoreJKS | Path of the truststore file for internal TLS connections. | Empty | TLS |
+|singlenode.enableAuth | Whether to enable authentication and authorization for clients. |False| Auth |
+|singlenode.passwdFile | Path of the file containing user credentials and ACLs, for the PasswordAuthHandler. |Empty| Auth |
+|singlenode.userName | The default username used for internal communication between Segment Store and Controller. | Empty| Auth |
+|singlenode.passwd | The default password used for internal communication between Segment Store and Controller. | Empty| Auth |
```
Comment (Aparnarr) [inline documentation/src/docs/security/pravega-security-configurations.md]: Parameters formats ,please change

File: documentation/src/docs/security/pravega-security-configurations.md (hunk 1 above)
Comment (ravisharda) [reply documentation/src/docs/security/pravega-security-configurations.md]: Done.

File: documentation/src/docs/security/pravega-security-configurations.md (hunk 3 above)
Comment (ravisharda) [reply documentation/src/docs/security/pravega-security-configurations.md]: Done.

File: documentation/src/docs/security/pravega-security-configurations.md (hunk 5 above)
Comment (ravisharda) [reply documentation/src/docs/security/pravega-security-configurations.md]: Done.

File: documentation/src/docs/security/pravega-security-configurations.md (hunk 4 above)
Comment (ravisharda) [reply documentation/src/docs/security/pravega-security-configurations.md]: Done.

File: documentation/src/docs/security/pravega-security-configurations.md (hunk 7 above)
Comment (ravisharda) [reply documentation/src/docs/security/pravega-security-configurations.md]: The phrase I used here "PEM-encoded server certificate" is more appropriate and precise in this context.

File: documentation/src/docs/security/pravega-security-configurations.md (hunk 8 above)
Comment (ravisharda) [reply documentation/src/docs/security/pravega-security-configurations.md]: The phrase I used here "PEM-encoded server certificate" is more appropriate and precise in this context. As for the term `Auth` in the cells, using a capital letter `A` is more appropriate as it is the first letter in the cell. Quite like `E` in "Empty" or `F` in "False" in other columns.

File: documentation/src/docs/security/pravega-security-configurations.md (hunk 10 above)
Comment (ravisharda) [reply documentation/src/docs/security/pravega-security-configurations.md]: Done.

File: documentation/src/docs/security/pravega-security-configurations.md (hunk 9 above)
Comment (ravisharda) [reply documentation/src/docs/security/pravega-security-configurations.md]: Done.

File: documentation/src/docs/security/pravega-security-configurations.md (hunk 2 above)
Comment (ravisharda) [reply documentation/src/docs/security/pravega-security-configurations.md]: Done.

File: documentation/src/docs/security/pravega-security-configurations.md (hunk 6 above)
Comment (ravisharda) [reply documentation/src/docs/security/pravega-security-configurations.md]: As described elsewhere, the existing phrase is better suited in this context.

**Pull request merged**